dissect/target/plugins/os/windows/powershell.py

```diff
@@ -35,21 +41,25 @@ def powershell_history(self):
         """Return PowerShell command history for all users.
 
         The PowerShell ConsoleHost_history.txt file contains information about the commands executed with PowerShell in
-        a terminal. No data is recorded from terminal-less PowerShell sessions.
+        a terminal. No data is recorded from terminal-less PowerShell sessions. Commands are saved to disk after the process has completed.
+        PSReadLine does not save commands containing 'password', 'asplaintext', 'token', 'apikey' or 'secret'.
 
         References:
             - https://0xdf.gitlab.io/2018/11/08/powershell-history-file.html
-        """
-        for user, path in self._history:
-            for line in path.open("r"):
+            - https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_history?view=powershell-7.3#order-of-commands-in-the-history
+            - https://learn.microsoft.com/en-us/powershell/module/psreadline/about/about_psreadline?view=powershell-7.3#command-history
+        """  # noqa E501
+
+        for user, _path in self._history:
+            for line in _path.open("r"):
                 line = line.strip()
                 if not line:
                     continue
 
                 yield ConsoleHostHistoryRecord(
-                    last_modified=path.stat().st_mtime,
+                    mtime=_path.stat().st_mtime,
                     command=line,
-                    path=uri.from_windows(str(path)),
+                    source=path.from_windows(str(_path)),
```

Comment:
```suggestion
                    source=_path,
```
I believe this should work.